src/core/src/UsdNotificationHandler.cpp

```diff
@@ -83,300 +38,453 @@ bool inSubtree(const pxr::SdfPath& path, const pxr::SdfPath& descendantPath) {
 
-    // clang-format off
     for (const auto& property : properties) {
-        if (detectTransformChanges && (property == pxr::CesiumTokens->cesiumAnchorDetectTransformChanges ||
-                                       property == pxr::UsdTokens->xformOp_transform_cesium)) {
-            updateByUsdTransform = true;
-        } else if (property == pxr::CesiumTokens->cesiumAnchorGeographicCoordinates) {
-            updateByLatLongHeight = true;
+        if (detectTransformChanges &&
+            (property == PXR_NS::CesiumTokens->cesiumAnchorDetectTransformChanges ||
+             property == PXR_NS::UsdTokens->xformOp_translate || property == PXR_NS::UsdTokens->xformOp_rotateXYZ ||
+             property == PXR_NS::UsdTokens->xformOp_scale)) {
+            updateByPrimLocalTransform = true;
+            updateBindings = true;
+        } else if (property == PXR_NS::CesiumTokens->cesiumAnchorGeographicCoordinates) {
+            updateByGeographicCoordinates = true;
+            updateBindings = true;
         } else if (
-            property == pxr::CesiumTokens->cesiumAnchorPosition ||
-            property == pxr::CesiumTokens->cesiumAnchorRotation ||
-            property == pxr::CesiumTokens->cesiumAnchorScale) {
-            updateByFixedTransform = true;
+            (adjustOrientation && property == PXR_NS::CesiumTokens->cesiumAnchorAdjustOrientationForGlobeWhenMoving) ||
+            property == PXR_NS::CesiumTokens->cesiumAnchorPosition ||
+            property == PXR_NS::CesiumTokens->cesiumAnchorRotation ||
+            property == PXR_NS::CesiumTokens->cesiumAnchorScale) {
+            updateByPrimLocalToEcefTransform = true;
+            updateBindings = true;
+        } else if (property == PXR_NS::CesiumTokens->cesiumAnchorGeoreferenceBinding) {
+            updateByGeoreference = true;
+            updateBindings = true;
         }
     }
-    // clang-format on
 
-    if (updateByUsdTransform) {
-        GeospatialUtil::updateAnchorByUsdTransform(origin.value(), globeAnchor);
-    } else if (updateByLatLongHeight) {
-        GeospatialUtil::updateAnchorByLatLongHeight(origin.value(), globeAnchor);
-    } else if (updateByFixedTransform) {
-        GeospatialUtil::updateAnchorByFixedTransform(origin.value(), globeAnchor);
+    if (updateByPrimLocalTransform) {
+        pGlobeAnchor->updateByPrimLocalTransform();
+    }
+
+    if (updateByGeographicCoordinates) {
+        pGlobeAnchor->updateByGeographicCoordinates();
+    }
+
+    if (updateByPrimLocalToEcefTransform) {
+        pGlobeAnchor->updateByPrimLocalToEcefTransform();
+    }
+
+    if (updateByGeoreference) {
+        pGlobeAnchor->updateByGeoreference();
+    }
+
+    if (updateBindings) {
+        updateGlobeAnchorBindings(pContext, globeAnchorPath);
     }
 }
 
-void processCesiumIonServerChanged(const pxr::SdfPath& ionServerPath, const std::vector<pxr::TfToken>& properties) {
+void processCesiumIonServerChanged(
+    Context* pContext,
```

Comment:
Pretty minor, but to use `const` more frequently this could be

```suggestion
    Context* const pContext,
```

Reply:
In general we don't really use const when passing by value (in this case the pointer is being passed by value).

However there could be a case for this being `const Context* pContext`.

Reply:
> However there could be a case for this being `const Context* pContext`.

Never mind, it needs be non-const in this function